### PR TITLE
fix: cleanup `messages` between files (`tree.messages`)

### PR DESCRIPTION
**Pull request by michael-ciniawsky**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -97,6 +97,31 @@ PostHTML.prototype.process = function (tree, options) {
...
+  /**
+  * Messages to store and pass metadata
+  *
+  * @memberof tree
+  *
+  * @type {Array} messages
+  *
+  * @example
+  * ```js
+  * export default function plugin (options = {}) {
```
Comment (voischev) [inline lib/index.js]: support node v0.10

Comment (michael-ciniawsky) [reply lib/index.js]: It's just JSDoc example 🙃 :) ? But I can use CJS (which might be better anyways)

**Comment by michael-ciniawsky** [comment]
If we have another way to fix this in a better way please chime in, since keeping messages on the API would be better by design. But I don't know how to do it in a clean way

File: lib/index.js (hunk 1 above)
Comment (michael-ciniawsky) [reply lib/index.js]: done

**Comment by GitScrum** [comment]
@michael-ciniawsky I created a [posthtml-api-message-demo](https://github.com/inside-demo/posthtml-api-message-demo) and it works for me, what does not work for you?

**Comment by michael-ciniawsky** [comment]
@GitScrum `messages` aren't 'cleaned' up between `posthtml.process()` calls atm within the same node process. The `api` {Object} is extended with the `messages`of the previous `process()` call

```js
posthtml.process(file1.html, options)
  .then(({ messages }) => {
      // [ { dependency: 'from file1.html' } ]
      console.log(messages)
  })

posthtml.process(file2.html, options)
  .then(({ messages }) => {
       // [ { dependency: 'from file1.html' } /* <=* */, { dependency: 'file2' } ]
      console.log(messages)
  })
```
* isn't a dependency of `file2.html` (Bug)

Someone correct my if I'm wrong please :), but I got this 'duplicates'  in `html-loader/posthtml-loader`. It makes sense in the way it's currently implemented and to be expected, but is undesired for integrations (gulp, webpack etc...). `messages` should be stored **per** `process()`  call (Plugins IPC) on the `api` object and then 'reseted' by `tree.messages = []` when calling `process()` (again).

**Comment by michael-ciniawsky** [comment]
Please gives this some priority it blocks the `html-loader@next` (`v1.0.0`) release atm 😞

**Comment by GitScrum** [comment]
@michael-ciniawsky I remember, please give me time to figure it out. Hope for understanding

**Comment by michael-ciniawsky** [comment]
@GitScrum https://github.com/posthtml/esm/blob/master/test/__snapshots__/import.test.js.snap#L34-L58  The first two `Object` are from the previous test running before in the same process

**Comment by GitScrum** [comment]
@michael-ciniawsky because you always passed a link to the `api` object

**Comment by michael-ciniawsky** [comment]
@GitScrum Yep exactly :)

**Pull request closed without merging**